google/cloud/storage/examples/storage_object_samples.cc

```diff
@@ -64,6 +64,18 @@ void InsertObject(gcs::Client client, int& argc, char* argv[]) {
 }
 //! [insert object]
 
+//! [get object metadata]
+void GetObjectMetadata(gcs::Client client, int& argc, char* argv[]) {
+  if (argc < 3) {
```

Comment:
I understand the reason for `argc`, `argv`, and `ConsumeArg()`, but these introduce more information into the sample than is necessary for a developer to know. 

Suggested sample design:
```cpp
void GetObjectMetadata(gcs::Client client, std::string bucket_name, std::string object_name) {
  // [START storage_object_metadata]
  // Shorter namespace
  // namespace gcs = google::cloud::storage;

  // Initialized storage client
  // gcs::Client client;

  // std::string bucket_name = "bucket-name";
  // std::string object_name = "prefix/to/object";

  auto metadata = client.GetObjectMetadata(bucket_name, object_name);
  std::cout << "The metadata is " << metadata << std::endl;
  // [END storage_object_metadata]
}
```

Reply:
And what about a compromise: I need the functions to have the same signature so the code is easy to maintain, but I see your point about reducing the noise, so what about this:

```C++
void GetObjectMetadataCommand(gcs::Client client, int& argc, char* argv[]) {
  // the weird stuff goes here...
  // [START get object metadata] 
  auto get_object_metadata_example = [](
      google::cloud::storage::Client, std::string bucket_name, std::string object_name) {
    auto metadata = client.GetObjectMetadata(bucket_name, object_name);
    std::cout << "The metadata is " << metadata << end::endl; 
  };
  // [END get object metadata]
  get_object_metadata_example();
}
```

With that the developer only sees this:

```C++
  auto get_object_metadata_example = [](
      google::cloud::storage::Client, std::string bucket_name, std::string object_name) {
    auto metadata = client.GetObjectMetadata(bucket_name, object_name);
    std::cout << "The metadata is " << metadata << end::endl; 
  };
```

Which is clear about the types that the API consumes, and does not introduce the weird `ConsumeArg()` or anything.

Does that work?

Reply:
Also, let me figure out if there is a way to mix both the `// [START ...] / [END ... ]` and the `//! [...]` markers.  With the `//!` doxygen markers the code can be surfaces into the docstrings [examples here](https://googlecloudplatform.github.io/google-cloud-cpp/latest/bigtable/classgoogle_1_1cloud_1_1bigtable_1_1v0_1_1Table.html)

Reply:
Understood, I appreciate the lambda over the existing sample. What do you think about the example below?  Thanks @coryan 

```cpp
auto get_object_metadata_example = \ 
// [START storage_object_metadata]
[](google::cloud::storage::Client client, std::string bucket_name, std::string object_name) {
    auto metadata = client.GetObjectMetadata(bucket_name, object_name);
    std::cout << "The metadata is " << metadata << end::endl; 
}
// [END storage_object_metadata]
```

Reply:
Thanks Carlos! SGTM

Reply:
I ran into a couple of problems:

- clang-format made a mess of the comments if we interleave them, I found an alternative that probably look reasonable, let me know what you think.

- using the `[START ..]` and `[END]` markers inside the Doxygen markers looks super ugly.  I imagine that the Doxygen markers would look ugly the other way too.  

Can you take a look and see if the thing I tried would work?

Reply:
After testing on cloud.google.com this is a suitable format.

Thanks @coryan LGTM.